homeassistant/components/remote/harmony.py

```diff
@@ -233,13 +247,15 @@ def new_activity(self, activity_info: tuple) -> None:
         self._current_activity = activity_name
         self._state = bool(activity_id != -1)
         self._available = True
-        self.async_schedule_update_ha_state()
+        if self._platform_added:
```

Comment:
Instead we should register the callbacks in `async_added_to_hass`. Can we register the callbacks after instantiating the client?

Reply:
It can be done afterwards. Do you want me to do that update in this PR then?

Reply:
Great! Yes, please! Then we don't need to add `_platform_added`.

Reply:
LOL. Figured you would say that. Working on it right now. :-)

Reply:
Done, tested here to make absolutely sure every callback still works (verifying I did not have a bug in aioharmony <g>).